2.5.4-alpha

Consti10 opened the pull request:
Requres updated openhd core !

Impoved param set(s):
1) added auto-fetch
2) added custom UI elements for changing camera, resolution
3) whitelisted even more params (more things are done by openhd automatically) for a clear, user friendly ui
4) removed V_ prefixes from param
5) no more V_OS_CAM_CONFIG ;)
6) more verbose view when air / ground is not alive (Icon)

Sidebar:
Adds a new UI element that is easier to use than the main configpopup, an joystick navigatable (for the future)

TCP:
New approach for auto connect.
Only activate on android though.


Other:
Make SOCStatus widget generic (remove duplicated code)
Add dev overlay (stats)
And a few more things ...

Consti10 committed:
add raw parse & decode

Consti10 committed:
add raw parse & decode

Consti10 committed:
fix weird malloc bug (hopefully) that crashes tcp connection from time to time. work on raw

Consti10 committed:
fix weird malloc bug (hopefully) that crashes tcp connection from time to time. work on raw

Consti10 committed:
fix weird malloc bug (hopefully) that crashes tcp connection from time to time. work on raw

Consti10 committed:
fix scale to fit grayed out on android

Consti10 committed:
back to rtp for now

Consti10 committed:
Improve Camera selection & camera (AIR CAM1, AIR CAM2) user interface
(Hide full param set)

Consti10 committed:
user interface- improve ui of param set when connection is lost, generify the (big) settings popup, work on camera selection

Consti10 committed:
user interface- improve ui of param set when connection is lost, generify the (big) settings popup, work on camera selection

Consti10 committed:
user interface- improve ui of param set when connection is lost, generify the (big) settings popup, work on camera selection

Consti10 committed:
more verbose camera param names

Consti10 committed:
improve connection mode auto / tcp connection (auto reconnect on dead server)

Consti10 committed:
improve connection mode auto / tcp connection (auto reconnect on dead server)

Consti10 committed:
experimental - add auto fetch params functionality

Consti10 committed:
improve params

Consti10 committed:
can't really remember all the things done ...

Consti10 committed:
can't really remember all the things done ...

Consti10 committed:
document a few more params

Consti10 committed:
fix a few comp warnings

Consti10 committed:
fix a few comp warnings

Consti10 committed:
fix a few comp warnings

Consti10 committed:
make soc status widget generic to reduce duplicated code

Consti10 committed:
add platform type, improve UI

Consti10 committed:
add dev overlay (stats), add preliminary joystick support for setting frequency, ...

Consti10 committed:
improve joystick menu, allow changing frequency on air only

Consti10 committed:
improve busy indicator, and joy

Consti10 committed:
use the quick panel as a new element only for joystick for now

Consti10 committed:
finalize joystick edit values element

Consti10 committed:
Merge remote-tracking branch 'origin/2.5-evo' into consti-dev

# Conflicts:
#	app/telemetry/MavlinkTelemetry.cpp

Consti10 committed:
fix icon of gnd (CPU) status widget

Consti10 committed:
Finalize autofetch of params

Consti10 committed:
uncomment Quick panel for now

Consti10 committed:
A ton of things:
1) async 'interface' for qml -> cpp param -> qml
2) Generic UI element for mavlink int-choices
3) Use this UI element in the sidebar where appropriate

Consti10 committed:
A ton of things:
1) async 'interface' for qml -> cpp param -> qml
2) Generic UI element for mavlink int-choices
3) Use this UI element in the sidebar where appropriate

Consti10 committed:
A ton of things:
1) async 'interface' for qml -> cpp param -> qml
2) Generic UI element for mavlink int-choices
3) Use this UI element in the sidebar where appropriate

Consti10 committed:
A ton of things:
1) async 'interface' for qml -> cpp param -> qml
2) Generic UI element for mavlink int-choices
3) Use this UI element in the sidebar where appropriate

Consti10 committed:
A ton of things:
1) async 'interface' for qml -> cpp param -> qml
2) Generic UI element for mavlink int-choices
3) Use this UI element in the sidebar where appropriate

Consti10 committed:
improve param

Consti10 committed:
add setting 5180Mhz / lowband for pretty much norbert only ;)

Consti10 committed:
sidebar generify

Consti10 committed:
sidebar generify - make it (almost ;) joystick navigatable, fix a few other issues

Consti10 committed:
add joystick navigation to sidebar

knputyi commented:
[> Requres updated openhd core !
> 
> Impoved param set(s):
> 
> 1. added auto-fetch
> 2. added custom UI elements for changing camera, resolution
> 3. whitelisted even more params (more things are done by openhd automatically) for a clear, user friendly ui
> 4. removed V_ prefixes from param
> 5. no more V_OS_CAM_CONFIG ;)
> 6. more verbose view when air / ground is not alive (Icon)
> 
> TCP: New approach for auto connect - TODO reduce wasted CPU cycles
> 
> Other: Make SOCStatus widget generic (remove duplicated code) Add dev overlay (stats) Added navigatable UI for goggles (not finished) And more ...](https://github.com/OpenHD/QOpenHD/pull/659/commits/a5eb8e8bffe9eaf1de4ba6e6f6d08a857607b2d1)

I'm using 5280MHz, not 5180... ;)

Consti10 committed:
add joystick navigation to sidebar

Consti10 committed:
finalize sidebar, for now ;) @Raphael

Consti10 committed:
add option to log the param set result (async)

Consti10 committed:
override qopenhd batt n cells setting from openhd if set

Consti10 committed:
fix a few minor quirks,show radiation icon instead of annoying tx errors log message

Consti10 committed:
finalize FC_BATT_N_CELLS param

Consti10 commented:
> [> Requres updated openhd core !
> 
> > Impoved param set(s):
> > 
> > 1. added auto-fetch
> > 2. added custom UI elements for changing camera, resolution
> > 3. whitelisted even more params (more things are done by openhd automatically) for a clear, user friendly ui
> > 4. removed V_ prefixes from param
> > 5. no more V_OS_CAM_CONFIG ;)
> > 6. more verbose view when air / ground is not alive (Icon)
> > 
> > TCP: New approach for auto connect - TODO reduce wasted CPU cycles
> > Other: Make SOCStatus widget generic (remove duplicated code) Add dev overlay (stats) Added navigatable UI for goggles (not finished) And more ...]([a5eb8e8](https://github.com/OpenHD/QOpenHD/commit/a5eb8e8bffe9eaf1de4ba6e6f6d08a857607b2d1))
> 
> I'm using 5280MHz, not 5180... ;) oh, hm, I need to change that then ;)

Consti10 committed:
lowband - use 40Mhz spacing, show everything in the range of 5180 - 5300

Consti10 committed:
Frequency element - implement disarm requirement (can be overridden)

Consti10 committed:
Generify - mavlink choices now also supports string, so we can use it for the reslution_fps

Consti10 committed:
default camera choices

Consti10 committed:
default camera resolution choice(s)

Consti10 committed:
default camera resolution choice(s)

Consti10 committed:
Fix the annoying TCP cpu usage - if we are running on the ground station
itself, we don't need all this autoconnect jam ;)

Consti10 committed:
Fix the annoying TCP cpu usage - if we are running on the ground station
itself, we don't need all this autoconnect jam ;)

Consti10 committed:
i disabled a few checks for testing way back here...

Consti10 committed:
update mavlink

Consti10 committed:
update mavlink

Consti10 committed:
cleanup, also replace moodycamel queue with simple thread safe queue using std::condition_variable

Consti10 committed:
cleanup, also replace moodycamel queue with simple thread safe queue using std::condition_variable

Consti10 committed:
cleanup, also replace moodycamel queue with simple thread safe queue using std::condition_variable

Consti10 committed:
generify rc channels element

Consti10 committed:
Sidebar - add flips. Add busy indicator for a restarting camera.
Threadsafe queue, optimize mutex congestion,too.

Consti10 committed:
fix the sidebar channels

Consti10 committed:
typos

Consti10 committed:
experiment - OSD widget presets. uncomment for now

Consti10 committed:
experiment - OSD widget presets. uncomment for now

Consti10 committed:
use proper model for resolutions inside sidebar

Consti10 committed:
use proper model for resolutions inside sidebar

Consti10 committed:
use proper model for resolutions inside sidebar

Consti10 committed:
use proper model for resolutions inside sidebar

Consti10 committed:
use proper model for resolutions inside sidebar

Consti10 committed:
fix weird binding loops

Consti10 committed:
fix ui bug where camera selection dialoque shown for rock is incorrect

Consti10 committed:
show not available if the param is not exposed for a setting (which means the HW does not support changing it)

Consti10 committed:
wb max fec blk size changed in openhd
